lib/active_record/connection_adapters/redshift_adapter.rb

```diff
@@ -391,7 +391,7 @@ class Table < ActiveRecord::ConnectionAdapters::Table
       ADAPTER_NAME = 'Redshift'
 
       NATIVE_DATABASE_TYPES = {
-        primary_key: "serial primary key",
+        primary_key: "integer identity",
```

Comment:
To chime in with some of my initial experiences here:

- The default `serial primary key` seems to cause `ActiveRecord::StatementInvalid: PG::FeatureNotSupported: ERROR:  Column "sequences.id" has unsupported type "serial".`  Why others don't get this, I don't know.

- By using `integer primary key` as suggested by @aamine , I get: `ActiveRecord::StatementInvalid: PG::InternalError: ERROR:  Cannot insert a NULL value into column id` - indicating that RDS is not auto-incrementing, and that perhaps I should be doing something for that application-side.  Not sure what is actually recommended in this case.

- With using @nunommc 's `integer identity`, I get `ActiveRecord::UnknownPrimaryKey: Unknown primary key for table lists in model List.`  This is solved by adding to my models: `self.primary_key = "id"`.